src/nginx/t/cloud_trace.t

```diff
@@ -137,29 +175,58 @@ is($json_obj->{traces}->[0]->{spans}->[0]->{parentSpanId}, $parent_span_id,
-    'Next trace span is CheckServiceControl');
+
+# span 1: FetchAccessToken
+is($json_obj->{traces}->[0]->{spans}->[1]->{name}, 'FetchAccessToken',
+    'Next trace span is FetchAccessToken');
 is($json_obj->{traces}->[0]->{spans}->[1]->{parentSpanId}, $rootid,
-    'Parent of CheckServiceControlCache span is root');
-my $check_service_control_id = $json_obj->{traces}->[0]->{spans}->[1]->{spanId};
-is($json_obj->{traces}->[0]->{spans}->[2]->{name}, 'CheckServiceControlCache',
+    'Parent of FetchAccessToken span is root');
+
+# span 2: CheckAuth
+is($json_obj->{traces}->[0]->{spans}->[2]->{name}, 'CheckAuth',
+    'Next trace span is CheckAuth');
+is($json_obj->{traces}->[0]->{spans}->[2]->{parentSpanId}, $rootid,
+    'Parent of CheckAuth span is root');
+my $check_auth_id = $json_obj->{traces}->[0]->{spans}->[2]->{spanId};
+
+# span 2: HttpFetch
+is($json_obj->{traces}->[0]->{spans}->[3]->{name}, 'HttpFetch',
+    'Next trace span is HttpFetch');
+is($json_obj->{traces}->[0]->{spans}->[3]->{parentSpanId}, $check_auth_id,
+    'Parent of HttpFetch span is CheckAuth');
+
+# span 4: CheckServiceControl
+is($json_obj->{traces}->[0]->{spans}->[4]->{name}, 'CheckServiceControl',
+    'Next trace span is CheckServiceControl');
+is($json_obj->{traces}->[0]->{spans}->[4]->{parentSpanId}, $rootid,
+    'Parent of CheckServiceControl span is root');
+my $check_service_control_id = $json_obj->{traces}->[0]->{spans}->[4]->{spanId};
+
+# span 5: CheckServiceControlCache
+is($json_obj->{traces}->[0]->{spans}->[5]->{name}, 'CheckServiceControlCache',
     'Next trace span is CheckServiceControlCache');
-is($json_obj->{traces}->[0]->{spans}->[2]->{parentSpanId}, $check_service_control_id,
+is($json_obj->{traces}->[0]->{spans}->[5]->{parentSpanId}, $check_service_control_id,
     'Parent of CheckServiceControlCache span is CheckServiceControl');
-my $check_service_control_cache_id = $json_obj->{traces}->[0]->{spans}->[2]->{spanId};
-is($json_obj->{traces}->[0]->{spans}->[3]->{name}, 'Call ServiceControl server',
+my $check_service_control_cache_id = $json_obj->{traces}->[0]->{spans}->[5]->{spanId};
+
+# span 6: Call ServiceControl server
+is($json_obj->{traces}->[0]->{spans}->[6]->{name}, 'Call ServiceControl server',
     'Next trace span is Call ServiceControl server');
-is($json_obj->{traces}->[0]->{spans}->[3]->{parentSpanId}, $check_service_control_cache_id,
+is($json_obj->{traces}->[0]->{spans}->[6]->{parentSpanId}, $check_service_control_cache_id,
     'Parent of Call ServiceControl sever span is CheckServiceControlCache');
-is($json_obj->{traces}->[0]->{spans}->[4]->{name}, 'QuotaControl',
-    'Next trace span is Backend');
-is($json_obj->{traces}->[0]->{spans}->[4]->{parentSpanId}, $rootid,
-    'Parent of Beckend span is root');
-is($json_obj->{traces}->[0]->{spans}->[5]->{name}, 'Backend',
+
+# span 7: QuotaControl
+is($json_obj->{traces}->[0]->{spans}->[7]->{name}, 'QuotaControl',
```

Comment:
Can we also trigger QuotaServiceControlCache?

Reply:
Done